src/strategies/base.ts

```diff
@@ -141,7 +141,7 @@ export abstract class BaseStrategy implements Strategy {
   }
 
   async getDefaultPackageName(): Promise<string | undefined> {
-    return '';
+    return this.packageName ?? '';
```

Comment:
I think I had intended for `getDefaultPackageName()` to only contain the logic of fetching from another source (like `package.json`).

WDYT about making a `getPackageName()` accessor like we have for `getComponent()` that does the `this.packageName ||= await getDefaultPackageName()` logic.